frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -1121,9 +1121,16 @@ function DataFrame({
             onSortColumn={
               isSortingEnabled
                 ? (direction: "asc" | "desc" | undefined) => {
-                    // Cell selection are kept on the old position,
-                    // which can be confusing. So we clear all cell selections before sorting.
-                    clearSelection(true, true)
+                    if (isRowSelectionActivated && isRowSelected) {
+                      // Keeping row selections when sorting columns is not supported at the moment.
+                      // So we need to clear the selected rows before we do the sorting (Issue #11345).
+                      // Maintain column selections as these are not impacted.
+                      clearSelection(false, true)
```

Comment:
Think it makes sense here to only clear row selections while maintaining column selections, but lmk if alternative thoughts